contracts/AnchoredPriceView.sol

```diff
@@ -0,0 +1,331 @@
+        }
+
+        // load usdc for using in loop to convert anchor prices to dollars
+        uint256 usdcPrice = anchor.getUnderlyingPrice(cUsdcAddress);
+
+        // Try to update the view storage
+        for (uint i = 0; i < symbols.length; i++) {
+            string memory symbol = symbols[i];
+            address tokenAddress = getCTokenAddress(symbol);
+            uint256 sourcePrice = priceData.getPrice(source, symbol);
+            uint256 anchorPrice = getAnchorPrice(tokenAddress, usdcPrice);
+
+            if (anchorPrice == 0 || tokenAddress == cUsdcAddress || tokenAddress == cUsdtAddress) {
+                emit PriceGuarded(symbol, sourcePrice, anchorPrice);
+            } else {
+                uint256 anchorRatioMantissa = sourcePrice * 100e16 / anchorPrice;
+                // Only update the view's price if the source price is within a bound, and it is a new median
+                if (anchorRatioMantissa <= upperBoundAnchorRatio && anchorRatioMantissa >= lowerBoundAnchorRatio) {
+                    // only update and emit event if the source price is new, otherwise do nothing
+                    if (_prices[symbol] != sourcePrice) {
+                        _prices[symbol] = sourcePrice;
+                        emit PriceUpdated(symbol, sourcePrice);
+                    }
+                } else {
+                    emit PriceGuarded(symbol, sourcePrice, anchorPrice);
+                }
+            }
+        }
+    }
+
+    /**
+     * @notice Returns price denominated in USD, with 6 decimals
+     * @dev If price was posted by source, return it. Otherwise, return anchor price converted through source ETH price.
+     */
+    function prices(string calldata symbol) external view returns (uint256) {
+        uint256 price = _prices[symbol];
+
+        if (price != 0) {
+            return price;
+        } else {
+            uint256 usdPerEth = _prices["ETH"];
+            uint256 ethPerToken = anchor.getUnderlyingPrice(getCTokenAddress(symbol));
+
+            // ethPerToken has 18 decimals since the usdt, usdc, wbtc tokens hit
+            // usdPerEth has 6 decimals
+            // scaling by 1e18 and dividing leaves 1e6, as desired
+            return mul(usdPerEth, 1e18) / ethPerToken;
+        }
+    }
+
+    /**
+     * @dev fetch price in eth from proxy and convert to usd price using anchor usdc price.
+     * @dev Anchor usdc price has 30 decimals, and anchor general price has 18 decimals, so multiplying 1e18 by 1e18 and dividing by 1e30 yields 1e6
+     */
+    function getAnchorPrice(address tokenAddress, uint256 usdcPrice) public view returns (uint256) {
+        if ( tokenAddress == cUsdcAddress || tokenAddress == cUsdtAddress )  {
+            // hard code to 1 dollar
+            return oneDollar;
+        }
+
+        uint priceInEth = anchor.getUnderlyingPrice(tokenAddress);
+        uint additionalScale;
+        if ( tokenAddress == cWbtcAddress ){
```

Comment:
Reference point for conversation below-- why doesn't additional scale come from the function below?

Reply:
yeah this is a slightly different additional scale, 
this one is scaling up so that dividing by USDC price ( 1e30 ) does not truncate to 0.

Yes naming is confusing and should be changed. I'm also looking at running the anchor through ETH instead so that we are only ever converting in one direction